tests/PersistentPageIdentifiersIntegrationTest.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\PersistentPageIdentifiers\Tests;
+
+use Title;
+use WikiPage;
+
+class PersistentPageIdentifiersIntegrationTest extends \MediaWikiIntegrationTestCase {
+
+	protected function createPageWithText( string $text = 'Whatever wikitext' ): WikiPage {
+		$page = $this->getServiceContainer()->getWikiPageFactory()->newFromTitle( $this->createUniqueTitle() );
+
+		$this->editPage( $page, $text );
+
+		return $page;
+	}
+
+	private function createUniqueTitle(): Title {
+		static $pageCounter = 0;
+		return Title::newFromText( 'PPITestPage' . ++$pageCounter );
```

Comment:
I'm wondering if there is a mismatch between the lifetime of PHP and when the DB state is reset. Might not be unique. But shrug: if it works, whatever

The ideal test system creates a new DB from scratch for each test (method) so there is no chance of interaction between tests. Easy to achieve when building something well designed and fast with SQLite in-memory, but we are building on top of MediaWiki, so probably cannot achieve that without doing an excessive amount of backflips.

Reply:
I just copied this from https://github.com/ProfessionalWiki/PageApprovals/blob/master/tests/PageApprovalsIntegrationTest.php, so no thought went into this.